lib/mongoid/fields.rb

```diff
@@ -731,8 +731,19 @@ def field_for(name, options)
         opts = options.merge(klass: self)
         type_mapping = TYPE_MAPPINGS[options[:type]]
         opts[:type] = type_mapping || unmapped_type(options)
-        unless opts[:type].is_a?(Class)
+        if !opts[:type].is_a?(Class)
           raise Errors::InvalidFieldType.new(self, name, options[:type])
+        else
+          invalid_bson_classes = [ BSON::Decimal128, BSON::Int32, BSON::Int64 ]
```

Comment:
extract this to a constant

Reply:
I am not opposed to this but I don't think this is necessary either.